Go/Go.sublime-syntax

```diff
@@ -86,6 +86,13 @@ variables:
 
   exponent: '[Ee][+-]?'
 
+  # Matches a digit with any number of numeric separators, while
+  # not allowing a numeric separator as the last or first character.
+  ddigits: (?:\d+(?:_\d+)*)
+
+  # Hexideciaml counterpart to ddigits.
+  hdigits: (?:\h+(?:_\h+)*)
```

Comment:
I just want to mention that `(?:_?\h+)+` might be dangerous, because if I understand it right, that regex is ambiguous in how to match consecutive numbers, see https://snyk.io/blog/redos-and-catastrophic-backtracking/. I believe `(?:_?\h)+` would do the same, but prevents backtracking.

Reply:
[From the docs](https://golang.org/ref/spec#Integer_literals):

> ...an underscore character _ may appear after a base prefix or between successive digits...

The `_?{{hdigits}}` is catching that second case where it can appear between the base prefix and the first digit.

Reply:
I think adding the `_?` to the beginning of the `hdigits` variable adds to readability. Implemented in 62696fef.